test/poolFixtures.hpp

```diff
@@ -687,4 +688,117 @@ TEST_P(umfPoolTest, pool_from_ptr_half_size_success) {
+                        sizeof(arg));
+        ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+        ASSERT_EQ(arg, i + 1);
+        ptrs.push_back(ptr);
+    }
+
+    for (auto &ptr : ptrs) {
+        umf_result_t umf_result = umfPoolFree(pool_get, ptr);
+        ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    }
+    ptrs.clear();
+    ret = umfCtlGet("umf.pool.by_handle.stats.alloc_count", pool_get, &arg,
+                    sizeof(arg));
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(arg, 0);
+
+    if (umf_test::isReallocSupported(pool_get)) {
+        for (int i = 0; i < 10; i++) {
+            void *ptr;
+            if (i % 2 == 0) {
+                ptr = umfPoolMalloc(pool_get, size);
+            } else {
+                ptr = umfPoolRealloc(pool_get, nullptr, size);
+            }
+            ASSERT_NE(ptr, nullptr);
+            ret = umfCtlGet("umf.pool.by_handle.stats.alloc_count", pool_get,
+                            &arg, sizeof(arg));
+            ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+            ASSERT_EQ(arg, i + 1);
+            ptrs.push_back(ptr);
+        }
+        for (auto &ptr : ptrs) {
+            ptr = umfPoolRealloc(pool_get, ptr, size * 2);
+            ASSERT_NE(ptr, nullptr);
+        }
+        ret = umfCtlGet("umf.pool.by_handle.stats.alloc_count", pool_get, &arg,
+                        sizeof(arg));
+        ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+        ASSERT_EQ(arg, num_allocs);
+        size_t allocs = ptrs.size();
+        for (auto &ptr : ptrs) {
+            if (allocs-- % 2 == 0) {
+                ptr = umfPoolRealloc(pool_get, ptr, 0);
+                ASSERT_EQ(ptr, nullptr);
+            } else {
+                ret = umfPoolFree(pool_get, ptr);
+                ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+            }
+        }
+        ptrs.clear();
+        ret = umfCtlGet("umf.pool.by_handle.stats.alloc_count", pool_get, &arg,
+                        sizeof(arg));
+        ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+        ASSERT_EQ(arg, 0);
+    }
+
+    if (umf_test::isCallocSupported(pool_get)) {
+        for (int i = 0; i < 10; i++) {
+            void *ptr = umfPoolCalloc(pool_get, 1, size);
+            ASSERT_NE(ptr, nullptr);
+            ret = umfCtlGet("umf.pool.by_handle.stats.alloc_count", pool_get,
+                            &arg, sizeof(arg));
+            ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+            ASSERT_EQ(arg, i + 1);
+            ptrs.push_back(ptr);
+        }
+
+        for (auto &ptr : ptrs) {
+            umf_result_t umf_result = umfPoolFree(pool_get, ptr);
+            ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+        }
+        ptrs.clear();
+        ret = umfCtlGet("umf.pool.by_handle.stats.alloc_count", pool_get, &arg,
+                        sizeof(arg));
+        ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+        ASSERT_EQ(arg, 0);
+    }
+
+    for (size_t i = 0; i < num_allocs; i++) {
+        void *ptr = umfPoolAlignedMalloc(pool_get, size, 4096);
```

Comment:
not supported on windows + use utils_get_page_size() (see umfPoolAlignedMalloc test)